sdk/samples/iot/paho_iot_pnp_with_provisioning_sample.c

```diff
@@ -250,13 +250,15 @@ static void register_device_with_provisioning_service(void)
   uint8_t mqtt_payload[MQTT_PAYLOAD_BUFFER_LENGTH];
   size_t mqtt_payload_length;
 
+  AZ_PUSH_IGNORE_DEPRECATIONS
```

Comment:
We may want to update our samples to no longer use deprecated APIs and provide alternatives that exist. Otherwise, customers will just copy/paste this.

Reply:
DSP v2 is not GA or publicly available yet. Moving to the new APIs will happen with the HSFM work.